### PR TITLE
tiva/lm3s6965-ek: Add module/elf definitions

### PR DESCRIPTION
**Pull request by yamt**
Basically a copy-and-paste from spresense.
But I added -mlong-calls for ELF to avoid out of range relocations.
(Isn't it necessary for spresense?)

### CODE REVIEW

File: boards/arm/tiva/lm3s6965-ek/scripts/gnu-elf.ld [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/arm/tiva/lm3s6965-ek/scripts/gnu-elf.ld]: All new files need to have a Apache 2.0 license.  Please copy-paste the header from sched/sched/sched_getcpu.c.

Comment (yamt) [reply boards/arm/tiva/lm3s6965-ek/scripts/gnu-elf.ld]: as stated in the commit message, this is a copy of an existing file. i'm not the right person to change the license.

Comment (yamt) [reply boards/arm/tiva/lm3s6965-ek/scripts/gnu-elf.ld]: @masayuki2009 can i replace this license to apache? this is a copy of spresense version.

**Comment by patacongo** [comment]
I am the copyright holder of the file that you copied.  I grant you permission to change the license header to apache 2.0.

This PR cannot come into the repository unless all new files have the Apache 2.0 license.

**Comment by patacongo** [comment]
Sorry, didn't mean to close that.  I hit the wrong button.

**Comment by yamt** [comment]
@patacongo 

the file's original copyright is:

> *   Copyright 2019 Sony Home Entertainment & Sound Products Inc.
> *   Author: Masayuki Ishikawa <Masayuki.Ishikawa@jp.sony.com>

**Comment by patacongo** [comment]
I wrote all of the ELF module linker scripts.  I don't know why my copyright was changed on this one.  It should not have been.  You can take one of the other versions fo the same file which has my copyright.  It cannot come in without an Apache 2.0  license.

**Comment by yamt** [comment]
@patacongo done. thank you.

**Pull request merged**